File: client/src/app/shared/Models/agreement.ts
export interface Landlord {
  id: number;
  keyNav: string;
  fullname: string;
  address: string;
  email: string;
  mobileNumber: string;
  photoUrl: string;
}

export interface Tenant {
  id: number;
  keyNav: string;
  fullName: string;
  mobileNumber: string;
  contactNumber: string;
  email: string;
  photoUrl: string;
  tenantNote: string;
}

export interface Agreement {
  id: number;
  keyNav: string;
  dateAgreement: Date;
  startDate: Date;
  endDate: Date;
  landlords: Landlord[];
  tenants: Tenant[];
  address: string;
  rent: number;
  premises: string;
}

File: BaseStartup/Entities/Agreement.cs
using Microsoft.EntityFrameworkCore;

namespace WebAPIHouses.Entities
{
    public class Agreement
    {
        public int Id { get; set; }
        public DateTime? DateAgreement { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public List<Landlord> Landlords { get; set; } = new();
        public List<Tenant> Tenants { get; set; } = new();
        public int HouseId { get; set; }
        public House? House { get; set; }
        [Precision(precision: 10, scale: 2)]
        public decimal Rent { get; set; }
        public string UserId { get; set; } = string.Empty;
    }
}